distribution/lib/Standard/Base/0.0.0-dev/src/Data/Set.enso

```diff
@@ -0,0 +1,106 @@
+import project.Any.Any
+import project.Data.Array_Proxy.Array_Proxy
+import project.Data.Map.Map
+import project.Data.Numbers.Integer
+import project.Data.Ordering.Comparable
+import project.Data.Ordering.Ordering
+import project.Data.Vector.Vector
+import project.Data.Text.Extensions
+import project.Data.Text.Text
+import project.Errors.Illegal_Argument.Illegal_Argument
+import project.Nothing.Nothing
+import project.Panic.Panic
+
+from project.Data.Boolean import Boolean, True, False
+
+## UNSTABLE
+   An unordered collection of unique values.
+type Set
+    ## PRIVATE
+    Value (underlying_map : Map Any Boolean)
+
+    ## Constructs a new set from a vector.
+
+       Arguments:
+       - vector: the vector of elements to add to the set.
+       - allow_duplicates: specifies if duplicate elements in the input are
```

Comment:
Maybe `allow_overwrite` would be clearer; duplicates suggests that this might be a multi-set.

Reply:
Very good point! I didn't think of it.

Reply:
I'm just not completely sure of `allow_overwrite` either - `overwrite` sounds just so much more mutable or file-related, at least to me.

What about `error_on_duplicates`? Then it is clear that it is a setting that controls whether this is an error, not that it allows multiset. Would that work?

Reply:
The text description of this parameter is wrong as its not overriding the value.
I think on the `Map`, @GregoryTravis's name is clearer so probably worth changing that one.

Reply:
What do you mean "its not overriding the value."?

In the meantime I pushed a commit with changing to `error_on_duplicates`. Will that work? Or shall I rename it to `allow_overwrite`? IMO `overwrite` is unclear in this context.

Reply:
`allow_update` perhaps, distinguishing insertion from update.